packages/workbox-build/src/entry-points/options/common-generate-schema.js

```diff
@@ -38,16 +38,25 @@ module.exports = baseSchema.keys({
       'staleWhileRevalidate'
     )],
     options: joi.object().keys({
-      cacheName: joi.string(),
-      plugins: joi.array().items(joi.object()),
-      cacheExpiration: joi.object().keys({
-        maxEntries: joi.number().min(1),
-        maxAgeSeconds: joi.number().min(1),
-      }).or('maxEntries', 'maxAgeSeconds'),
+      backgroundSync: joi.object().keys({
+        name: joi.string().required(),
+        options: joi.object(),
+      }),
+      broadcastCacheUpdate: joi.object().keys({
+        channelName: joi.string().required(),
+        options: joi.object(),
+      }),
       cacheableResponse: joi.object().keys({
         statuses: joi.array().items(joi.number().min(0).max(599)),
         headers: joi.object(),
       }).or('statuses', 'headers'),
+      cacheName: joi.string(),
+      expiration: joi.object().keys({
+        maxEntries: joi.number().min(1),
+        maxAgeSeconds: joi.number().min(1),
+      }).or('maxEntries', 'maxAgeSeconds'),
+      networkTimeoutSeconds: joi.number().min(1),
+      plugins: joi.array().items(joi.object()),
```

Comment:
How does this work? I.e. how does code in the node space get into the service worker? My main consideration here is that if this works, why support one way in the build process and a different way in the service worker?

Reply:
The code that converts whatever is configured in the `node` build tools into a string representation that could be inserted into a service worker file is

https://github.com/GoogleChrome/workbox/blob/v3/packages/workbox-build/src/lib/runtime-caching-converter.js

and the string is used in the template at

https://github.com/GoogleChrome/workbox/blob/dafc65ba54f4f320e89df13d3e8a3ee2ed710662/packages/workbox-build/src/templates/sw-template.js#L62

(None of that has changed recently.)

The syntax used to generate the service worker code is not exactly the same as the syntax used when executing the service worker runtime. Service worker execution implies that we're inside of the `ServiceWorkerGlobalScope` with the various definitions in the `workbox` namespace available. When the build configuration is used, we're inside of the `node` runtime environment, and we maintain a mapping of the keys in the config that correspond to the resulting runtime code.

I think in this PR the mapping between configuration supplied to `node` and resulting Workbox v3 service worker runtime code gets much closer to being 1:1, as we removed some differences that were mainly around to match `sw-precache`'s config.